contracts/Certificate.sol

```diff
@@ -91,14 +91,18 @@ contract Certificate is
    * @param certificateAmount The total number of NRTs retired in this certificate.
    * @param removalIds The removal IDs used for the certificate.
    * @param removalAmounts The amounts from each removal used for the certificate.
+   * @param purchasingTokenAddress The address of the token used to purchase the certificate.
+   * @param priceMultiple The number of purchasing tokens required to buy one NRT.
    */
   event ReceiveRemovalBatch(
     address from,
     address indexed recipient,
     uint256 indexed certificateId,
     uint256 certificateAmount,
     uint256[] removalIds,
-    uint256[] removalAmounts
+    uint256[] removalAmounts,
+    address purchasingTokenAddress,
+    uint256 priceMultiple
```

Comment:
so, very weirdly, I cannot seem to generate a new snapshot for this branch.  no errors, all the tests run when I run `forge snapshot`, but there are no differences in the generated .gas-snapshot .... going to have to figure this out, let me know if you have any ideas

Reply:
oh, this is because there was a failing test (which I believe has been failing since[ this PR](https://github.com/nori-dot-eco/contracts/pull/435)) and when I comment out that test the snapshot runs

Reply:
so I tried packing the event args more tightly by moving the addresses to be all together, which means those three args can fit in two words, and the gas actually went up slightly from the original snapshot.  the difference was super tiny.  In what sense can you see the purchasing token in the originating transaction?  there are permit sig components but nothing from which the purchasing token contract address can be derived.  I suppose that contract is going to emit `TransferFrom` events that we could comb through....

I'm fine with removing the purchasing token contract address if you feel confident that there's a straightforward path to reconstructing this information in whatever analytics we will need to build

Reply:
@scott any opinions?

Reply:
@amiecorso I'd prefer to keep it.  This will be a critical part of reports built on the market transaction flow and the additional headache of trying to correlate this event to all possible source tokens doesn't seem worth it.

Reply:
agreed.  I also realized in some later comments that I think jaycen might be thinking of the old ERC777 system which would indeed have very obvious information about the purchasing token because the originating transaction would interact with that contract.  In this case, the only link we'd have to the ERC20 would be the transfer events it would emit as part of that transaction and i'm not even sure those would have the token address.